### ISSUE TITLE
Abandon  Vundle / Pathogen / NeoBundle / vim-plug and conform to the new standard Vim 8  pack commands

### ISSUE THREAD

**Issue by zingus** [rust-lang/rust.vim#137]
Vundle / Pathogen / NeoBundle / vim-plug all have issues, some are a pain to install, the others have very non-intuitive command names. With Vim 8 now offering a standard packaging mechanism, it's silly to suggest new users to install those.

**Comment by steveklabnik** [comment]
I like this idea, at least, in supporting it, but until Vim 8 is widely available, we'd still need to leave some support. I don't know what vim version availability is like.

**Comment by Pratyush** [comment]
Vim also has a separate plugin interface from NeoVim IIRC; the plugin managers allow you to use the plugin on both. I'd like if the manager support was possible somehow.

**Comment by chris-morgan** [comment]
Vim 8’s package mechanism basically just has you put rust.vim repository in ~/.vim/pack/whatever-you-want/start/rust.vim and then it all works. Really, that’s it.

**Comment by romanows** [comment]
I resisted installing this for some time because installing a Vim package manager (and even choosing among the 4 (!) currently listed in the README) was a bit intimidating.  I ended up independently coming across the method noted by @chris-morgan, and think it could be helpful to add in the README.

```
mkdir -p ~/.vim/pack/rust.vim/start/rust.vim
git clone https://github.com/rust-lang/rust.vim.git ~/.vim/pack/rust.vim/start/rust.vim
```

I'd make a pull request for this but I don't know enough about Vim to say that this is equivalent to using a package manager and I haven't used the plugin enough yet to be sure that it is working correctly.  Looking forward to finally trying it out, though, thanks!

**Comment by di3663** [comment]
@romanows  thanks for let me know. I really do not like those plugins.

I use and only use one plugin `NERDTree`, hope rust.vim helps.

**Comment by zingus** [comment]
"How to install rust.vom on Vim 8" installation should simply be featured at the top of the README. Old package managers are silly, and people should not be incouraged to mess with them, as it is easier and more productive to upgrade to Vim 8 than to install and understand one of those relics.

Windows Install Instructions

    mkdir "%HOMEDRIVE%%HOMEPATH%\vimfiles\pack\rust.vim\start\rust.vim"
    git clone https://github.com/rust-lang/rust.vim.git "%HOMEDRIVE%%HOMEPATH%\vimfiles\pack\rust.vim\start\rust.vim"

OR, as package managing allows different collections of packages to be swapped around, with a package collection with the same name as the user...

    mkdir "%HOMEDRIVE%%HOMEPATH%\vimfiles\pack\%USERNAME%\start\rust.vim"
    git clone https://github.com/rust-lang/rust.vim.git "%HOMEDRIVE%%HOMEPATH%\vimfiles\pack\%USERNAME%\start\rust.vim"

...that would make the linux install instructions look like this:

    mkdir -p ~/.vim/pack/$USER/start/rust.vim
    git clone https://github.com/rust-lang/rust.vim.git ~/.vim/pack/$USER/start/rust.vim

**Comment by pickfire** [comment]
What about the update instructions?

**Comment by abusch** [comment]
I'm really not impressed by the OP's tone. Pretty much every Vim plugin out there has instructions for various package managers. I agree it's not ideal, and if one of them wins the race and is supported on both vim and neovim, then great, but in the meantime, there's not much point going around pretending they don't exist and telling everyone they're "silly" for using those "relics" or making snarky remarks about "being open to change".

**Comment by zingus** [comment]
@abusch a major version happened. Rust.vim works out of the box on it. The README doesn't reflect that. This is the issue.

I'm a normal rust user. I'm not into the Vim 8 / Neovim wars. In fact I originally knew nothing about them and I'd preferred to keep it that way. I came from Vim 7, looked into the package managers available for it, got sick of them, trying to understand the slight minutiae that differentiate one from the other, trying to understand which one I needed, if I had to commit to one or what... and anyway updated because I already wanted to update to the next major version of vim. I've lost possibly a couple of days trying to understand what I needed to do to get rust.vim going. I would have preferred to devote that time into learning more rust. The editor shouldn't get in the way of a learning a language.

**Comment by da-x** [comment]
@zingus I agree that the README.md is currently massively outdated. However I also think that rust.vim should cater to the contemporary conventions of both the Rust and Vim worlds, which means that on the present state where most Vim users still rely heavily on plugin managers, we would need to provide instructions for them (at least for those plugin managers that are actively maintained - I am in favor of removing instructions regarding NeoBundle, for example).

README.md overhaul is on the To Do. I'd keep this open until it's done.
